kroxylicious-app/src/assembly/run-java.sh

```diff
@@ -0,0 +1,690 @@
+  # Check whether -XX:TieredStopAtLevel is already given in JAVA_OPTIONS
+  if echo "${JAVA_OPTIONS:-}" | grep -q -- "-XX:TieredStopAtLevel"; then
+    return
+  fi
+  if [ $(c2_disabled) = true ]; then
+    echo "-XX:TieredStopAtLevel=1"
+  fi
+}
+
+# Switch on diagnostics except when switched off
+diagnostics_options() {
+  if [ -n "${JAVA_DIAGNOSTICS:-}" ]; then
+    if [ "${JAVA_MAJOR_VERSION:-0}" -ge "11" ]; then
+      echo "-XX:NativeMemoryTracking=summary -Xlog:gc*:stdout:time -XX:+UnlockDiagnosticVMOptions"
+    else
+      echo "-XX:NativeMemoryTracking=summary -XX:+PrintGC -XX:+PrintGCDateStamps -XX:+PrintGCTimeStamps -XX:+UnlockDiagnosticVMOptions"
+    fi
+  fi
+}
+
+# Replicate thread ergonomics for tiered compilation.
+# This could ideally be skipped when tiered compilation is disabled.
+# The algorithm is taken from:
+# OpenJDK / jdk8u / jdk8u / hotspot
+# src/share/vm/runtime/advancedThresholdPolicy.cpp
+ci_compiler_count() {
+  local core_limit="$1"
+  local log_cpu=$(calc 'log2($1)' "$core_limit")
+  local loglog_cpu=$(calc 'log2(max2($1,1))' "$log_cpu")
+  local count=$(calc 'max2($1*$2,1)*3/2' "$log_cpu" "$loglog_cpu")
+  local c1_count=$(calc 'max2($1/3,1)' "$count")
+  local c2_count=$(calc 'max2($1-$2,1)' "$count" "$c1_count")
+  [ $(c2_disabled) = true ] && echo "$c1_count" || echo $(calc '$1+$2' "$c1_count" "$c2_count")
+}
+
+cpu_options() {
+  # JVMs >= 10 know about CPU limits
+  if [ "${JAVA_MAJOR_VERSION:-0}" -ge "10" ]; then
+    return
+  fi
+
+  local core_limit="${JAVA_CORE_LIMIT:-}"
+  if [ "$core_limit" = "0" ]; then
+    return
+  fi
+
+  if [ -n "${CONTAINER_CORE_LIMIT:-}" ]; then
+    if [ -z ${core_limit} ]; then
+      core_limit="${CONTAINER_CORE_LIMIT}"
+    fi
+    echo "-XX:ParallelGCThreads=${core_limit} " \
+         "-XX:ConcGCThreads=${core_limit} " \
+         "-Djava.util.concurrent.ForkJoinPool.common.parallelism=${core_limit} " \
+         "-XX:CICompilerCount=$(ci_compiler_count $core_limit)"
+  fi
+}
+
+#-XX:MinHeapFreeRatio=20  These parameters tell the heap to shrink aggressively and to grow conservatively.
+#-XX:MaxHeapFreeRatio=40  Thereby optimizing the amount of memory available to the operating system.
+heap_ratio() {
+  echo "-XX:MinHeapFreeRatio=20 -XX:MaxHeapFreeRatio=40"
+}
+
+# These parameters are necessary when running parallel GC if you want to use the Min and Max Heap Free ratios.
+# Skip setting gc_options if any other GC is set in JAVA_OPTIONS.
+# -XX:GCTimeRatio=4
+# -XX:AdaptiveSizePolicyWeight=90
+gc_options() {
+  if echo "${JAVA_OPTIONS:-}" | grep -q -- "-XX:.*Use.*GC"; then
+    return
+  fi
+
+  local opts=""
+  # for JVMs < 10 set GC settings
+  if [ -z "${JAVA_MAJOR_VERSION:-}" ] || [ "${JAVA_MAJOR_VERSION:-0}" -lt "10" ]; then
+    opts="${opts} -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 $(heap_ratio)"
+  fi
+  if [ -z "${JAVA_MAJOR_VERSION:-}" ] || [ "${JAVA_MAJOR_VERSION:-}" != "7" ]; then
+    opts="${opts} -XX:+ExitOnOutOfMemoryError"
```

Comment:
I'd like to see ExitOnOutOfMemoryError be on by default.  If we've gone OOM, exiting in the only reasonable course of action.

Reply:
Oh I see it is already.